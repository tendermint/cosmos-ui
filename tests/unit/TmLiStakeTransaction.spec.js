import { mount } from "@vue/test-utils"
import { TmLiStakeTransaction } from "@/index"
import transactions from "../../assets/txs.js"

describe("TmLiStakeTransaction", () => {
  let wrapper
  let propsData = {
    transaction: transactions[3],
    validators: [
      {
        owner: "cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw",
        moniker: "cool validator"
      }
    ],
    validatorURL: "/validator"
  }

  beforeEach(() => {
    wrapper = mount(TmLiStakeTransaction, { propsData })
  })

  it("should show delegations", () => {
    expect(wrapper.vm.delegation).toBe(true)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it("should show unbondings", () => {
    wrapper.setProps({
      transaction: transactions[4]
    })
    expect(wrapper.vm.unbonding).toBe(true)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
