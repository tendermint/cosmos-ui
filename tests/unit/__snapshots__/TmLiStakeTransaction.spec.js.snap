// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmLiStakeTransaction should show delegations 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #512584;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <p
      class="tm-li-tx__content__caption"
    >
      <div>
        Delegated 
        <b>
          4200
        </b>
        <span>
           steak
        </span>
      </div>
    </p>
    <div
      class="tm-li-tx__content__information"
    >
      <div
        class="tm-li-tx__content__information__details"
      >
        <div>
          To 
          <a
            href="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
          >
            cool validator
          </a>
        </div>
      </div>
      <div
        class="tm-li-tx__content__information__block"
      >
        <a
          href="https://explorecosmos.network/blocks/566"
        >
          Block #566 
        </a>
        @ 2:03 am
      </div>
    </div>
  </div>
</div>
`;

exports[`TmLiStakeTransaction should show unbondings 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #9d5c19;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <p
      class="tm-li-tx__content__caption"
    >
      <div>
        Unbonded 
        <b>
          100
        </b>
        <span>
           Steak
        </span>
      </div>
    </p>
    <div
      class="tm-li-tx__content__information"
    >
      <div
        class="tm-li-tx__content__information__details"
      >
        <div>
          From 
          <a
            href="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
          >
            cool validator
          </a>
        </div>
      </div>
      <div
        class="tm-li-tx__content__information__block"
      >
        <a
          href="https://explorecosmos.network/blocks/566"
        >
          Block #566 
        </a>
        @ 2:03 am
      </div>
    </div>
  </div>
</div>
`;
